### DISCUSSION TITLE
Optionally disable DNS in specific WLAN by SSID (or MAC)

### DISCUSSION THREAD

**Discussion by DarthSonic** [celzero/rethink-app#635] [Ideas]
I would like to be able to disable Rethink App DNS resolver in a specific WLAN automatically.

**Comment by ignoramous** [comment]
> I would like to be able to disable Rethink App DNS resolver in a specific WLAN automatically.

I'd imagine you can use Tasker for that, but Rethink, today, doesn't *natively* integrate with it: #63 

I must say, even though this feature is *trivial* to implement, it comes at a cost: Android requires that apps request "coarse location" permission to read *SSID*s (which makes sense since *SSID*s and *MAC*s are pretty damning to privacy, and not many people spoof it even though they can with no real consequence).

See also: #342

**Reply by ignoramous** [reply]
Separate app? That's the bit that Tasker might help with, no? We probably need to integrate with Tasker (but haven't).

**Comment by SunnyWebeSimile** [comment]
Almost 3 years after this discussion arose, the Location permission was added to Rethink in `v055u`.

I have a use case that would benefit greatly from the requested feature. I have a recursive DNS resolver and a self-hosting server on my LAN, and would like to reach that server using its domain name at all times—**including when I am connected to my LAN**. I have port-forwarded the necessary services to allow Internet access, and my LAN's DNS points my domain name to the server's private IP address to allow intranet access.

However, I would also like to use encrypted DNS on ny Android phone. I am unwilling to port-forward my recursive DNS resolver for this.

Android's built-in DoT is a no-go. I cannot use my LAN's DNS when it is on, and there are more private services I do not want to port-forward that I am thus unable to access before turning off Private DNS on my phone. Rethink is an app I already use for its VPN-based IP firewall, and it would be the perfect solution for my phone, but I would also have to open Rethink frequently and switch its upstream DNS accordingly. Due to this inconvenience, I have found myself not using encrypted DNS at all.

Rethink being able to automatically select its upstream DNS server **(or even just switch to Firewall-only mode)** based on Wi-Fi SSIDs would remove this barrier and allow me to secure my DNS again. Please consider this feature enhancement for your next update.

**Reply by ignoramous** [reply]
Sure, please see:

- #1130
